source/clone.js

```diff
@@ -3,9 +3,9 @@ import _curry1 from './internal/_curry1';
 
 
 /**
- * Creates a deep copy of the value which may contain (nested) `Array`s and
- * `Object`s, `Number`s, `String`s, `Boolean`s and `Date`s. `Function`s are
- * assigned by reference rather than copied
+ * Creates a deep copy of the source that can be used in place of the source object without retaining any references to it.
+ * The source object may contain (nested) `Array`s, `Object`s, `Number`s, `String`s, `Boolean`s and `Date`s.
+ * `Function`s are assigned by reference rather than copied.
  *
```

Comment:
Do we want an additional, more explicit paragraph here?  Perhaps something like:

>      * Note that if the source object has multiple nodes that share a reference,
>      * the returned object will have the same structure, but the data the new 
>      * references point to will be a *copy* of the original.